src/netstandard/Microsoft.ServiceFabric.Actors.KVSToRCMigration/DataValidationWorker.cs

```diff
@@ -0,0 +1,400 @@
+
+            try
+            {
+                var startSN = this.workerInput.StartSeqNum;
+                if (this.workerInput.LastAppliedSeqNum.HasValue)
+                {
+                    startSN = this.workerInput.LastAppliedSeqNum.Value + 1;
+                    if (startSN > this.workerInput.EndSeqNum)
+                    {
+                        using (var tx = this.stateProvider.GetStateManager().CreateTransaction())
+                        {
+                            await this.CompleteWorkerAsync(tx, cancellationToken);
+                            WorkerResult tresult = await GetResultAsync(this.metadataDict, tx, this.Input.Phase, this.Input.Iteration, this.Input.WorkerId, this.traceId, cancellationToken);
+                            await tx.CommitAsync();
+
+                            return tresult;
+                        }
+                    }
+                }
+
+                var endSN = startSN + this.migrationSettings.ItemsPerEnumeration - 1;
+                long keysMigrated = 0L;
+
+                while (startSN <= this.workerInput.EndSeqNum)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+                    if (endSN > this.workerInput.EndSeqNum)
+                    {
+                        endSN = this.workerInput.EndSeqNum;
+                    }
+
+                    keysMigrated += await this.FetchAndSaveAsync(startSN, this.migrationSettings.ItemsPerEnumeration, endSN, cancellationToken);
+                    startSN = endSN + 1;
+                    endSN += this.migrationSettings.ItemsPerEnumeration;
+                }
+
+                var result = await this.GetResultAsync(cancellationToken);
+                ActorTrace.Source.WriteInfoWithId(
+                           TraceType,
+                           this.traceId,
+                           $"Completed migration worker\n Result: {result.ToString()} ");
+
+                return result;
+            }
+            catch (Exception ex)
+            {
+                ActorTrace.Source.WriteErrorWithId(
+                            TraceType,
+                            this.traceId,
+                            $"Migration worker failed with error: {ex} \n Input: /*Dump input*/");
+
+                throw ex;
+            }
+        }
+
+        private async Task<WorkerResult> GetResultAsync(CancellationToken cancellationToken)
+        {
+            WorkerResult result;
+            using (var tx = this.stateProvider.GetStateManager().CreateTransaction())
+            {
+                result = await GetResultAsync(this.metadataDict, tx, this.Input.Phase, this.Input.Iteration, this.Input.WorkerId, this.traceId, cancellationToken);
+                await tx.CommitAsync();
+            }
+
+            return result;
+        }
+
+        private async Task<long> FetchAndSaveAsync(long startSN, long snCount, long endSN, CancellationToken cancellationToken)
```

Comment:
Are you enumerating by sequence number here? I assume we should enumerate based on list of keys input.